lms/static/sass/course/modules/_student-notes.scss

```diff
@@ -215,6 +215,33 @@ $notes-annotator-background-dark: rgba(122,122,122,0.6); // taken from annotator
 // highlight
 .annotator-hl {
   background: $student-notes-highlight-color-focus;
+
+  // CASE: a highlight contains a link
+  a {
+    color: $link-color;
+
+    // STATE: hover/focus
+    &:hover, &:focus {
+      text-decoration: underline;
+    }
+
+    // STATE: disabled
+    &:disabled, &.is-disabled {
+      @extend %ui-disabled;
+      opacity: 0.5;
+      cursor: not-allowed;
+    }
+  }
+}
+
+// CASE: a link contains a highlight
```

Comment:
CASE 2: link in content has note's ``<span>`` injected into its DOM (as a child). This is the more common case by the vendor code.